### PR TITLE
Add viewport property to plotly pane to hold the axis range for all axes

### PR DESCRIPTION
**Pull request by jonmmease**
This PR is a follow on to the initial plotly callback support added in https://github.com/pyviz/panel/pull/529.

This adds a new `viewport` property to the plotly pane that holds the current axis ranges of all 2D cartesian axes in the figure.  This information isn't alway available through the `relayout_data` callback, for example when a figure is first displayed. This will be required to implement `Range*` streams in HoloViews, and by extension to support Datashader

Also added `viewport_update_policy` to control whether to update viewport
property with "continuous", "mouseup", or "throttle" policies.  When `viewport_update_policy='throttle'`, the throttle duration is controled by the new `viewport_update_throttle` property.  Here's an example of it in action.

```python
import panel as pn
import plotly.graph_objects as go
from plotly.subplots import make_subplots
pn.extension('plotly')

fig = make_subplots(rows=1, cols=2)
fig.add_scatter(y=[1, 3, 2], row=1, col=1);
fig.add_bar(y=[3, 2, 10], row=1, col=2);

fig_pane = pn.pane.Plotly(fig)
viewport_pane = pn.pane.Str()

pn.layout.Column(
    fig_pane,
    fig_pane.param.viewport,
    fig_pane.param.viewport_update_policy,
    fig_pane.param.viewport_update_throttle,
)
```
viewport_plotly

The `viewport` property can also be set from Python to set the figure viewport.

SetViewport

With this, the viewports of multiple views of the same pane will stay in sync.

```python
pn.layout.Column(fig_pane, fig_pane)
```

viewports_sync

I added `lodash` as a dependecy of the Plotly pane so handle deep object equality and throttling.  Including lodash will also make it easier to port more of the `FigureWidget` logic to the panel pane in the future.

### CODE REVIEW

**Comment by jonmmease** [comment]
Don't merge yet, there's a feedback bug that can cause updates to get stuck in an infinite loop when multiple views are moved together.  Investigating...

**Comment by jonmmease** [comment]
Testing again with #553 and there is still a problem.  The runaway feedback loop has gone away, but after panning one plot for a few seconds the other stops responding and I get the following error in Chrome:

Screen Shot 2019-07-28 at 4 22 44 PM

**Comment by jonmmease** [comment]
Alright, this is working reliable now 🎉  No more feedback loops or console errors when dragging in continuous mode when multiple views of the same figure are displayed together.

@philippjfr ready for review whenever you have time.  There are a couple of other tweaks I want to make to the plotly pane, but I'll do that in another PR since this one is pretty well self-contained now.

File: panel/pane/plotly.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ class Plotly(PaneBase):
     hover_data = param.Dict(doc="""hover callback data""")
     clickannotation_data = param.Dict(doc="""clickannotation callback data""")
     selected_data = param.Dict(doc="""selected callback data""")
+    viewport = param.Dict(doc="""current viewport state""")
+    viewport_update_policy = param.Selector(
+        objects=["continuous", "mouseup", "throttle"])
```
Comment (philippjfr) [inline panel/pane/plotly.py]: Should add a docstring and make the default explicit.

**Comment by philippjfr** [comment]
One minor comment, otherwise I'm happy to merge.

**Comment by jonmmease** [comment]
Thanks for taking a look. docs and defaults added.

**Pull request merged**